workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,462 @@
+      # For example to perform transformations on a web.config using the web.production.config file:
+      # dotnet publish --configuration Release /p:EnvironmentName=Production
+      - name: Build and Publish
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          dotnet build --configuration Release
+          dotnet publish -c Release -o ./published_app --no-restore
+
+      - name: Zip the published site for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          (cd published_app && zip -r ../${{ env.DEPLOY_ZIP }} .)
+
+      - name: Stop App Pool
+        id: app-pool-stop
+        if: always()
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'stop'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Deploy deployment package
+        if: steps.app-pool-stop.outcome == 'success'
+        uses: 'im-open/deploy-on-prem-web-app@v1.0.0'
+        with:
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          source-zip-file-path: ../${{ env.DEPLOY_ZIP }}
+          deployment-folder-path: ${{ needs.set-vars.outputs.WEBSITE_PATH }}
+          clean-deployment-folder: 'true'
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Start App Pool
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'start'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Close the PagerDuty Maintenance Window
+        if: always() && steps.open-window.outcome == 'success'
+        uses: im-open/close-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{ secrets.PAGERDUTY_API_KEY }}
+          maintenance-window-id: ${{ steps.open-window.outputs.maintenance-window-id }}
+
+      - name: Delete .zip and appsettings file that contains sensitive info
+        continue-on-error: true
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          rm -f ${{ env.DEPLOY_ZIP }}
+          rm -rf published_app
+          rm -f appsettings*.json
```

Pre-built artifact change:
```suggestion
      - name: Delete .zips and folder that contains sensitive info
        continue-on-error: true
        working-directory: ${{ env.PROJECT_ROOT }}
        run: |
          rm -f ${{ env.ASSET_ZIP }}
          rm -f ${{ env.DEPLOY_ZIP }}
          rm -rf ${{ env.UNZIPPED_ASSET }}
```